packages/contracts-bedrock/test/scripts/L2Genesis.t.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Test } from "forge-std/Test.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { VmSafe } from "forge-std/Vm.sol";
+
+import { Artifacts } from "scripts/Artifacts.s.sol";
+import { DeployConfig, DEV_ACCOUNT_FUND_AMT } from "scripts/DeployConfig.s.sol";
+import { Executables } from "scripts/Executables.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L2GenesisHelpers } from "scripts/libraries/L2GenesisHelpers.sol";
+import { L2Genesis } from "scripts/L2Genesis.s.sol";
+
+/// @notice Reads a `genesis-l2.json` file, parses the `alloc`s, and runs assertions
+///         against each alloc depending on whether it's a precompile, predeploy proxy,
+///         or predeploy implementation.
+contract L2Genesis_Test is Test, Artifacts {
+    DeployConfig public constant cfg =
+        DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
+
+    string internal genesisPath;
+    L2Genesis internal l2Genesis = new L2Genesis();
```

Comment:
It would be better to use `vm.etch` with the contract as it has no constructor. This will help to speed up compile time as the `L2Genesis` contract is large, it will mean that that its entire initcode will need to be included in the initcode of the test contract which means the initcode for every contract that is deployed with `CREATE` in `L2Genesis` will also be included. This makes it very slow to compile